File: hey_cortona/model/questions.py
from dataclasses import dataclass
from typing import Dict, List

from model.question import Question


@dataclass
class Questions:
    qid: str
    questions: List[Question]
    answers: Dict[str, float]

    @classmethod
    def from_mongo(cls, mongo_questions: Dict):
        return cls(mongo_questions["qid"], mongo_questions["questions"], mongo_questions["answers"])

    def __str__(self):
        return f"{self.qid}:{self.questions} - answers: {self.answers}"

File: questions_editor_server/models/Questions.js
const mongoose = require("mongoose");

let questionsSchema = new mongoose.Schema(
  {
    qid: {
      type: String,
    },
    questions: {
      type: [],
      require: false,
    },
    answers: {
      type: Object,
      require: true,
      default: {},
    },
  },
  { minimize: false }
);

module.exports = mongoose.model("question", questionsSchema);
